aws/resource_aws_eks_cluster.go

```diff
@@ -236,7 +245,32 @@ func resourceAwsEksClusterUpdate(d *schema.ResourceData, meta interface{}) error
 
 		err = waitForUpdateEksCluster(conn, d.Id(), updateID, d.Timeout(schema.TimeoutUpdate))
 		if err != nil {
-			return fmt.Errorf("error waiting for EKS Cluster (%s) update (%s): %s", d.Id(), updateID, err)
+			return fmt.Errorf("error waiting for EKS Cluster (%s) version update (%s): %s", d.Id(), updateID, err)
+		}
+	}
+
+	if d.HasChange("vpc_config") {
```

Comment:
I'm not sure why this is not acting like it seems it should, be needed to change this to check the nested attributes. 🙁 

```go
if d.HasChange("vpc_config.0.endpoint_private_access") || d.HasChange("vpc_config.0.endpoint_public_access") {
```

Previous output from acceptance testing:

```
--- FAIL: TestAccAWSEksCluster_Version (2519.24s)
   testing.go:538: Step 2 error: Error applying: 1 error occurred:
                * aws_eks_cluster.test: 1 error occurred:
                * aws_eks_cluster.test: error updating EKS Cluster (tf-acc-test-zdu6h) config: InvalidParameterException: Cluster is already at the desired configuration with endpointPrivateAccess: false and endpointPublicAccess: true
```

Even though the acceptance test debug logs were correctly only showing:

```
UPDATE: aws_eks_cluster.test
  version: "1.10" => "1.11"
```

Output from acceptance testing:

```
--- PASS: TestAccAWSEksClusterDataSource_basic (1320.87s)

--- PASS: TestAccAWSEksCluster_basic (1295.89s)
--- PASS: TestAccAWSEksCluster_VpcConfig_SecurityGroupIds (1327.26s)
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPrivateAccess (1490.80s)
--- PASS: TestAccAWSEksCluster_VpcConfig_EndpointPublicAccess (1526.63s)
--- PASS: TestAccAWSEksCluster_Version (2552.56s)
```

Reply:
I think it's a `schema.TypeList` vs. `schema.TypeSet` thing for the `vpc_config` attribute maybe?